registry/remote/repository.go

```diff
@@ -1318,6 +1332,10 @@ func (r *Repository) isReferrersAPIAvailable(ctx context.Context, desc ocispec.D
 		r.SetReferrersCapability(true)
 		return true, nil
 	case http.StatusNotFound:
+		if err := errutil.ParseErrorResponse(resp); errcode.IsErrorCode(err, errcode.ErrorCodeNameUnknown) {
+			// repository not found
+			return false, fmt.Errorf("%s: %w", ref.Repository, errdef.ErrNotFound)
```

Comment:
I'm wondering should we return error here.
`pingReferrersAPI()` is currently called by `Push()` and manifest `Delete()`.

1. For `Push()`, this is called AFTER pushing the manifest.
2. For `Delete()` this is called BEFORE deleting the manifest.

That is, the repository must be existing.